packages/gestalt/src/PageHeaderComponents.js

```diff
@@ -0,0 +1,212 @@
+// @flow strict
+import { useState, useRef, Fragment, type Element, type Node } from 'react';
+import Box from './Box.js';
+import Heading from './Heading.js';
+import Badge from './Badge.js';
+import Tooltip from './Tooltip.js';
+import IconButton from './IconButton.js';
+import Text from './Text.js';
+import Link from './Link.js';
+import Flex from './Flex.js';
+import Mask from './Mask.js';
+import Image from './Image.js';
+import Dropdown from './Dropdown.js';
+import { type ActionType } from './PageHeader.js';
+
+export function PageHeaderTitle({
+  hasThumbnail,
+  title,
+}: {|
+  hasThumbnail: boolean,
+  title: string,
+|}): Node {
+  return (
+    <Fragment>
+      <Box display="block" smDisplay="none">
+        <Heading size="400" lineClamp={1} accessibilityLevel={1}>
+          {title}
+        </Heading>
+      </Box>
+      <Box
+        display="none"
+        smDisplay="block"
+        dangerouslySetInlineStyle={{ __style: hasThumbnail ? { marginTop: '-5px' } : {} }}
```

Comment:
This seems like a leaky abstraction: this component is too aware of the context it's used in. Consider moving this margin to the markup surrounding PageHeaderTitle, which reduces the scope of this component and allows removing the hasThumbnail prop.

Reply:
Thumbnails are hidden on small breakpoints, so I need to apply the margin only on that case.